src/Schema/Exception/IncomparableNames.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Schema\Exception;
+
+use Doctrine\DBAL\Schema\Name\OptionallyQualifiedName;
+use Doctrine\DBAL\Schema\SchemaException;
+use LogicException;
+
+use function sprintf;
+
+final class IncomparableNames extends LogicException implements SchemaException
+{
+    public static function fromOptionallyQualifiedNames(
+        OptionallyQualifiedName $name1,
+        OptionallyQualifiedName $name2,
+    ): self {
+        return new self(sprintf(
+            'Non-equally qualified names are incomparable: %s, %s.',
```

Comment:
Are both strings guaranteed to be non-empty? Otherwise I would quote the %s.

Reply:
They are, and can also be quoted if the identifiers are quoted.

I couldn’t come up with any reasonable way to format names in exception messages, so this approach is used consistently in all exceptions.